plugins/ingest-script/src/main/java/org/elasticsearch/ingest/script/ScriptProcessor.java

```diff
@@ -0,0 +1,121 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.ingest.script;
+
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.ingest.core.AbstractProcessor;
+import org.elasticsearch.ingest.core.AbstractProcessorFactory;
+import org.elasticsearch.ingest.core.IngestDocument;
+import org.elasticsearch.node.Node;
+import org.elasticsearch.script.CompiledScript;
+import org.elasticsearch.script.Script;
+import org.elasticsearch.script.ScriptContext;
+import org.elasticsearch.script.ScriptService;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static java.util.Collections.emptyMap;
+import static org.elasticsearch.ingest.core.ConfigurationUtils.readOptionalStringProperty;
+import static org.elasticsearch.ingest.core.ConfigurationUtils.readStringProperty;
+import static org.elasticsearch.script.ScriptService.ScriptType.FILE;
+import static org.elasticsearch.script.ScriptService.ScriptType.INLINE;
+import static org.elasticsearch.script.ScriptService.ScriptType.STORED;
+
+public class ScriptProcessor extends AbstractProcessor {
+
+    public static final String TYPE = "script";
+    private final String field;
+    private final Script script;
+    private final ScriptService scriptService;
+    private final ClusterService clusterService;
+
+    public ScriptProcessor(String tag, String field, Script script, ScriptService scriptService,
+                           ClusterService clusterService) {
+        super(tag);
+        this.field = field;
+        this.script = script;
+        this.scriptService = scriptService;
+        this.clusterService = clusterService;
+    }
+
+    @Override
+    public void execute(IngestDocument ingestDocument) throws Exception {
+        Map<String, Object> data = new HashMap<>();
+        Map<String, Object> context = new HashMap<>();
+        data.put("ctx", context);
+        context.put("_source", ingestDocument.getSourceAndMetadata());
+        CompiledScript compiledScript = scriptService.compile(script, ScriptContext.Standard.INGEST, emptyMap(), clusterService.state());
+        Object returnValue = scriptService.executable(compiledScript, data).run();
+        ingestDocument.setFieldValue(field, returnValue);
+    }
+
+    @Override
+    public String getType() {
+        return TYPE;
+    }
+
+    public static final class Factory extends AbstractProcessorFactory<ScriptProcessor> {
+
+        private Node node;
+
+        public Factory(Node node) {
+            this.node = node;
+        }
+
+        @Override
+        protected ScriptProcessor doCreate(String processorTag, Map<String, Object> config) throws Exception {
+            String field = readStringProperty(TYPE, processorTag, config, "field");
```

Comment:
this `readStringProperty`? it is not parsing script code, it is an ingest-node config reader